### DISCUSSION TITLE
Use GDScript-style return type in function prototypes in the integrated docs

### DISCUSSION THREAD

**Discussion by mieldepoche** [godotengine/godot-proposals#7607] [Editor]
I can't unsee it, methods return types are C-style (`type function()`) but arguments are GDScript-style (`arg: Type`).
Does it make it easy to catch the return type at a glance? Does it date from a time where GDScript didn't support specifying return types?
I think it would be better to use the GDScript syntax for the GDScript docs.

|current|new|
|-|-|
| ![image](https://github.com/godotengine/godot-proposals/assets/71937306/d38b7994-04da-4e2f-ad51-24229a6c2dca) |![image](https://github.com/godotengine/godot-proposals/assets/71937306/6197b682-e6b5-4743-8aac-2387e0e533c5)|

I guess it makes sense in the methods summary, but idk...
![image](https://github.com/godotengine/godot-proposals/assets/71937306/1e17ebc7-1d0e-49d7-880c-f37b4b3bf6b0)

**Comment by dalexeev** [comment]
I think return type alignment in the table is more important.

**Reply by timothyqiu** [reply]
Maybe keep the return type as separate column in the table, but use `->` in the function description header?

**Comment by Calinou** [comment]
In my experience, the C-style signature style is more universally understood overall, and it's more relevant for C# users (and most GDExtension users). I'd keep it as-is for that reason.
